### PR TITLE
Add bash completion support

### PR DESCRIPTION
**Pull request by rhatdan**
I have taken the docker command completions code and reworked it for runc
support.

Signed-off-by: Dan Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by jfrazelle** [comment]
Cool!!!!!

On Wednesday, May 11, 2016, Daniel J Walsh <notifications@github.com> wrote:

> I have taken the docker command completions code and reworked it for runc
> support.
>
> Signed-off-by: Dan Walsh dwalsh@redhat.com
> <javascript:_e(%7B%7D,'cvml','dwalsh@redhat.com');>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/opencontainers/runc/pull/817
> Commit Summary
>
>    - Add bash completion support
>
> File Changes
>
>    - *M* Makefile
>    <https://github.com/opencontainers/runc/pull/817/files#diff-0> (9)
>    - *A* completions/bash/runc
>    <https://github.com/opencontainers/runc/pull/817/files#diff-1> (700)
>
> Patch Links:
>
>    - https://github.com/opencontainers/runc/pull/817.patch
>    - https://github.com/opencontainers/runc/pull/817.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/opencontainers/runc/pull/817>
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,8 @@
 	    localtest localunittest localintegration \
 	    test unittest integration
 
+PREFIX := $(DESTDIR)/usr
```
Comment (hqhq) [inline Makefile]: Should this be `$(DESTDIR)/usr/local`?

Comment (rhatdan) [inline Makefile]: Well I am not crazy about installing it by default in /usr/local. But if others want it there, I can put it there.

**Comment by crosbymichael** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (hqhq) [inline Makefile]: @rhatdan According to [Filesystem Hierarchy Standard](http://refspecs.linuxfoundation.org/FHS_2.3/fhs-2.3.pdf)
Software placed in / or /usr may be overwritten by system upgrades. For this reason, local software must
not be placed outside of /usr/local without good reason.

As a local software, I think we should stick runc binary in `/usr/local`.

Comment (rhatdan) [inline Makefile]: Changed default installation back to /usr/local/sbin/runc

**Comment by rhatdan** [comment]
@hqhq @mrunalp PTAL

**Comment by mrunalp** [comment]
LGTM

File: completions/bash/runc [hunk 2]
```diff
@@ -0,0 +1,700 @@
+#!/bin/bash
```
Comment (cyphar) [inline completions/bash/runc]: I think this file should live in `contrib/completions/bash`.

File: Makefile [hunk 3]
```diff
@@ -62,14 +64,16 @@ localintegration: all
 	bats -t tests/integration${TESTFLAGS}
 
 install:
-	install -D -m0755 runc /usr/local/sbin/runc
+	install -D -m0755 runc $(BINDIR)/runc
+	install -D -m0644 completions/bash/runc $(PREFIX)/share/bash-completion/completions/runc
```
Comment (cyphar) [inline Makefile]: Not all users have bash (I don't, for example). This should either be a separate install target (like `install-man`, we could have `install-bash`) or be a separate makefile in `contrib/completions/bash`.

**Comment by LK4D4** [comment]
+1 to both @cyphar suggestions
Would you mind to change @rhatdan ?

**Comment by rhatdan** [comment]
No problem.  Changed to use install-bash.  Also everything is now defaulting to /usr/local.

File: completions/bash/runc (hunk 2 above)
Comment (LK4D4) [inline completions/bash/runc]: @rhatdan what about this?

Comment (rhatdan) [inline completions/bash/runc]: moved

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**